### PR TITLE
revert cma when uninstall globalhub

### PR DESCRIPTION
**Pull request by ldpliu**
## Summary
This pr revert the cma when uninstall globalhub
We also redefine the placement and addondeploymentconfig to disable the sync managedhubcluster label to clusterclaim 
## Related issue(s)

Fixes #
https://issues.redhat.com/browse/ACM-14313
## Tests
* [x] Unit/function tests have been added and incorporated into `make unit-tests`.
* [x] Integration tests have been added and incorporated into `make integration-test`.
* [ ] E2E tests have been added and incorporated into `make e2e-test-all`.
* [x] List other manual tests you have done.

-   Manually tested the whole flow about importing hosted cluster and uninstall mgh

### CODE REVIEW

File: operator/pkg/controllers/hubofhubs/webhook/manifests/managedclustersetbinding.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{{- if .ImportClusterInHosted}}
+apiVersion: cluster.open-cluster-management.io/v1beta2
+kind: ManagedClusterSetBinding
+metadata:
+  name: global
```
Comment (clyang82) [inline operator/pkg/controllers/hubofhubs/webhook/manifests/managedclustersetbinding.yaml:5]: may use `global-hub-global`

File: operator/pkg/controllers/hubofhubs/webhook/manifests/placement.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+{{- if .ImportClusterInHosted}}
+apiVersion: cluster.open-cluster-management.io/v1beta1
```
Comment (clyang82) [inline operator/pkg/controllers/hubofhubs/webhook/manifests/placement.yaml:2]: who will use this `placement`?

File: operator/pkg/controllers/hubofhubs/webhook/manifests/addondeploymentconfig.yaml [hunk 3]
```diff
@@ -8,4 +8,7 @@ metadata:
     cluster.open-cluster-management.io/backup: globalhub
 spec:
   agentInstallNamespace: open-cluster-management-global-hub-agent-addon
+  customizedVariables:
+  - name: enableSyncLabelsToClusterClaims
```
Comment (clyang82) [inline operator/pkg/controllers/hubofhubs/webhook/manifests/addondeploymentconfig.yaml:12]: Do you need to update the `go.mod` to include the latest changes from foundation components?

File: operator/pkg/controllers/hubofhubs/webhook/manifests/managedclustersetbinding.yaml (hunk 1 above)
Comment (clyang82) [reply operator/pkg/controllers/hubofhubs/webhook/manifests/managedclustersetbinding.yaml:5]: how to use this new `ManagedClusterSetBinding`?

Comment (ldpliu) [reply operator/pkg/controllers/hubofhubs/webhook/manifests/managedclustersetbinding.yaml:5]: The clustersetbinding name must be same as clusterset. So it must be global.
this ManagedClusterSetBinding is used by placement. The placement must have the ManagedClusterSetBinding resource.

File: operator/pkg/controllers/hubofhubs/webhook/manifests/addondeploymentconfig.yaml (hunk 3 above)
Comment (ldpliu) [reply operator/pkg/controllers/hubofhubs/webhook/manifests/addondeploymentconfig.yaml:12]: foundation already have this feature, it just fix [some bug ](https://github.com/stolostron/multicloud-operators-foundation/pull/779)in there code. no need to change our go.mod.

File: operator/pkg/controllers/hubofhubs/webhook/manifests/placement.yaml (hunk 2 above)
Comment (ldpliu) [reply operator/pkg/controllers/hubofhubs/webhook/manifests/placement.yaml:2]: It used in patching cma:
https://github.com/stolostron/multicluster-global-hub/pull/1114/files#diff-451886b97b70c3cc3af9d6fa645359440d89d4e57401216152cefb77be036e3aR52

**Review by clyang82 — APPROVED** [review]
LGTM

**Pull request merged**